packages/client-search/src/types/Rule.ts

```diff
@@ -11,6 +11,11 @@ export type Rule = {
    */
   readonly condition?: Condition;
 
+  /**
+   * Conditions of the rule, expressed using the following variables: pattern, anchoring, context.
+   */
+  readonly conditions?: readonly Condition[];
```

Comment:
It should be a disjunct union: `{ condition } | { conditions }` with required in either case. That way you can’t have an object with both condition & conditions

Reply:
@Haroenv None of them are actually required.

However, we normally don't go that far with types. What do you think @Ant-hem @aseure?

Reply:
What does a query rule without condition mean? One of them is required, no?

Reply:
@Haroenv No. You can the only required files are: `objectID`, and `consequence`. This is quite often used by people that want to apply a query rule all the time.

If you agree, I would to prefer to keep the rule typing just simple like this. We usually don't go that far with types, like possible combinations of key in rules, etc.

Reply:
Agree with @nunomaduro on this: we don't go that far as enforcing the mutual exclusion here since the engine does not either.